blues/moves.py

```diff
@@ -177,70 +178,304 @@ def move(self, context):
+
+    def _findDart(self, nc_context):
+        """
+        Helper function to dynamically update dart positions based on positions
+        of other particles.
+        Arguments
+        ---------
+        basis_particles: list of 3 ints
+            Specifies the 3 indices of particles whose coordinates will be used
+            as basis vectors. If None is specified, uses those found in basis particles
+        Returns
+        -------
+        dart_list list of 1x3 np.arrays in units.nm
+            new dart positions calculated from the particle_pairs
+            and particle_weights
+        """
+        basis_particles = self.basis_particles
+        #make sure there's an equal number of particle pair lists
+        #and particle weight lists
+        dart_list = []
+        state_info = nc_context.getState(True, True, False, True, True, False)
+        temp_pos = state_info.getPositions(asNumpy=True)
+        part1 = temp_pos[basis_particles[0]]
+        part2 = temp_pos[basis_particles[1]]
+        part3 = temp_pos[basis_particles[2]]
+        for dart in self.n_dartboard:
+            old_center = self._findOldCoord(part1, part2, part3, dart)
+            dart_list.append(old_center)
+        self.dartboard = dart_list[:]
+        return dart_list
+
+    def _reDart(self, changevec):
+        """
+        Helper function to choose a random dart and determine the vector
+        that would translate the COM to that dart center
+        """
+        dartindex = np.random.randint(len(self.dartboard))
+        dvector = self.dartboard[dartindex]
+        chboard = dvector + changevec
+        return chboard
+
+    def _changeBasis(self, a, b):
+        '''
+        Changes positions of a particle (b) in the regular basis set to
+        another basis set (a).
+        Arguments
+        ---------
+        a: 3x3 np.array
+            Defines vectors that will create the new basis.
+        b: 1x3 np.array
+            Defines position of particle to be transformed into
+            new basis set.
+        Returns
+        -------
+        changed_coord: 1x3 np.array
+            Coordinates of b in new basis.
+        '''
+        ainv = np.linalg.inv(a.T)
+        changed_coord = np.dot(ainv,b.T)*unit.nanometers
+        return changed_coord
+
+    def _undoBasis(self, a, b):
+        '''
+        Transforms positions in a transformed basis (b) to the regular
+        basis set.
+        Arguments
+        ---------
+        a: 3x3 np.array
+            Defines vectors that defined the new basis.
+        b: 1x3 np.array
+            Defines position of particle to be transformed into
+            regular basis set.
+        Returns
+        -------
+        changed_coord: 1x3 np.array
+            Coordinates of b in new basis.
+        '''
+        a = a.T
+        changed_coord = np.dot(a,b.T)*unit.nanometers
+        return changed_coord
+
+
+    def _normalize(self, vector):
```

Comment:
`numpy.linalg.norm` is a somewhat misleading name. It'll give you the magnitude, but you'll still have to divide the vector to get the normalized vector you're looking for.